homeassistant/components/zwave_me/config_flow.py

```diff
@@ -0,0 +1,79 @@
+"""Config flow to configure ZWaveMe integration."""
+
+import logging
+
+from url_normalize import url_normalize
+import voluptuous as vol
+
+from homeassistant import config_entries
+
+from . import get_uuid
+from .const import CONF_TOKEN, CONF_URL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ZWaveMeConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """ZWaveMe integration config flow."""
+
+    def __init__(self):
+        """Initialize flow."""
+        self.url = vol.UNDEFINED
+        self.token = vol.UNDEFINED
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user or started with zeroconf."""
+        errors = {}
+        if self.url == vol.UNDEFINED:
+            schema = vol.Schema(
+                {
+                    vol.Required(CONF_URL): str,
+                    vol.Required(CONF_TOKEN): str,
+                }
+            )
+        else:
+            schema = vol.Schema(
+                {
+                    vol.Required(CONF_TOKEN): str,
+                }
+            )
+
+        if user_input is not None:
+            if self.url == vol.UNDEFINED:
+                self.url = user_input["url"]
+
+            self.token = user_input["token"]
+            if not self.url.startswith(("ws://", "wss://")):
+                self.url = f"ws://{self.url}"
+            self.url = url_normalize(self.url, default_scheme="ws")
+            if "://" not in self.url:
+                errors[CONF_URL] = "invalid_url"
+                self.url = vol.UNDEFINED
+                return self.async_show_form(
+                    step_id="user",
+                    data_schema=schema,
+                    errors=errors,
+                )
+            uuid = await get_uuid(self.url, self.token)
+            if uuid is not None:
+                await self.async_set_unique_id(uuid + self.url)
+                self._abort_if_unique_id_configured()
+                return self.async_create_entry(
+                    title=self.url,
+                    data={"url": self.url, "token": self.token},
+                )
+            errors[CONF_URL] = "could_not_retrieve_data"
+        return self.async_show_form(
+            step_id="user",
+            data_schema=schema,
+            errors=errors,
+        )
+
+    async def async_step_zeroconf(self, discovery_info):
+        """
+        Handle a discovered Z-Wave accessory - get url to pass into user step.
+
+        This flow is triggered by the discovery component.
+        """
+        self.url = discovery_info.host
+        return await self.async_step_user()
```

Comment:
We need to set the unique_id for the flow before leaving this step.

Reply:
Getting uuid for the unique_id requires token input by user. What should we do?

Reply:
There's no uuid or similar data in the discovery info?

The last resort is to fall back to a discovery without explicit unique id, which will mean some limitations for discovery.

https://github.com/home-assistant/core/blob/a0cd29bbcf761005114a58fedb04ec2bf97a6465/homeassistant/config_entries.py#L1343-L1351

Reply:
No, there is only an URL to distinguish between two Z-Wave-Me installations, so no valid unique data.
Will use this method and set proper unique_id after token input.

Reply:
@lawfulchaos /system/first-access API works without a token, so technically it can be used before the user entered the token